code/modules/vehicles/mecha/combat/greyscale/greyscale_constructor.dm

```diff
@@ -397,6 +402,8 @@ GLOBAL_LIST_INIT(greyscale_weapons_data, generate_greyscale_weapons_data())
 
 	mech.pixel_y = 240
 	animate(mech, time=4 SECONDS, pixel_y=initial(mech.pixel_y), easing=SINE_EASING|EASE_OUT)
+	if(!istype(src, /obj/machinery/computer/mech_builder/valhalla))
+		RegisterSignal(mech, COMSIG_PARENT_QDELETING, .proc/lock, override = TRUE)
```

Comment:
Don't istype src.
IMO just make the lock timer a var on the machine, then set it to something low on the valhalla version.